src/hooks/keyboard-navigation.js

```diff
@@ -47,33 +69,40 @@ function isElementVisible(element) {
  * [2] https://www.w3.org/TR/wai-aria-practices/#keyboard
  *
  * @param {import('preact').RefObject<HTMLElement>} containerRef
- * @param {object} options
- *   @param {boolean} [options.autofocus] - Whether to focus the first element
- *     in the set of matching elements when the component is mounted
- *   @param {boolean} [options.horizontal] - Enable navigating elements using left/right arrow keys
- *   @param {boolean} [options.vertical] - Enable navigating elements using up/down arrow keys
- *   @param {string} [options.selector] - CSS selector which specifies the
- *     elements that navigation moves between
+ * @param {KeyboardNavigationOptions} options
  */
-export function useArrowKeyNavigation(
+function useKeyboardNavigation(
   containerRef,
   {
+    active = true,
     autofocus = false,
-    horizontal = true,
-    vertical = true,
-    selector = 'a,button',
+    autofocusRef,
+    horizontal = false,
+    tab = false,
+    vertical = false,
+    selector = 'button, [href], input, select, textarea, [tabindex]',
```

Comment:
Some alternative ways I can think of to handle focus capture:

1. Use the native `<dialog>` element and its `showModal()` API. This is now supported in all browsers, but was only recently added to Firefox and Safari (~ March 2022). This involves the least amount of code, but needs a polyfill.
2. Listen for `focus` events on the document and cancel them if they would cause an element outside of the dialog to be focused. Google's [polyfill for dialog](https://github.com/GoogleChrome/dialog-polyfill/blob/master/index.js) does this. It also listens for `keydown` events on the document to detect if the user pressed Tab vs Shift+Tab just before the focus event.

Reply:
@robertknight If we went the native `dialog`-plus-polyfill route, what's your thinking on how to manage that polyfill? Would it be the responsibility of this package to register and provide this polyfill (and/or any other polyfills that come up in our journeys pertaining to common UI components)? Do you have a rough diagram in your mind?

Reply:
If we used https://github.com/GoogleChrome/dialog-polyfill#usage, that includes an ESM build that can be imported like any other module. I think we could then use it something like this:

```js
import dialogPolyfill from 'dialog-polyfill';
...
function MyDialog() {
  const dialogRef = useRef();
  useLayoutEffect(() => {
    dialogPolyfill.registerDialog(dialogRef.current);
  }, []);
  return <dialog ref={dialogRef}>...</dialog>
}

Reply:
This polyfill is quite small so I think we can get away without loading it conditionally. Another way to load it might be to have a function which gets called the first time that a component using `<dialog>` gets rendered, in order to install a global polyfill that monitors whenever `<dialog>` elements are added to the page. However I tend to prefer making polyfills work like any other code as far as possible (in terms of how they are imported and called), except that they defer to a native implementation if available.